File: mobile/app/src/main/java/lk/sliit/microgrid/data/model/Roles.kt
/*
 * File: Roles.kt
 * Project: Smart Solar Microgrid Trading System (SE4040)
 * Author: Lakshan
 * Created: 2026-09-22
 * Description: Role names used by the app. The spelling matches the Roles class in the API
 *              so the two never drift apart.
 */

package lk.sliit.microgrid.data.model

/**
 * Role names stored on a user and sent back by the API at login.
 */
object Roles {
    const val BACKOFFICE = "Backoffice"
    const val GRID_OPERATOR = "GridOperator"
    const val PROSUMER = "Prosumer"
}

File: web/src/roles.js
/*
 * File: roles.js
 * Project: Smart Solar Microgrid Trading System (SE4040)
 * Author: Lakshan
 * Created: 2026-09-22
 * Description: Role names and the home page each role lands on after login. The spelling
 *              matches the Roles class in the API so the two never drift apart.
 */

export const ROLES = {
  BACKOFFICE: 'Backoffice',
  GRID_OPERATOR: 'GridOperator',
  PROSUMER: 'Prosumer'
}

/**
 * Returns the page a role should be sent to after logging in.
 * Prosumers have no web pages - they use the Android app - so they get null.
 */
export function homePathForRole(role) {
  if (role === ROLES.BACKOFFICE) {
    return '/backoffice'
  }

  if (role === ROLES.GRID_OPERATOR) {
    return '/operator'
  }

  return null
}
